showcase/compliance_test.go

```diff
@@ -0,0 +1,137 @@
+		t.Fatalf(err.Error())
+	}
+
+	for _, group := range suite.GetGroup() {
+		rpcsToTest := group.GetRpcs()
+		for requestIdx, requestProto := range group.GetRequests() {
+			for rpcIdx, rpcName := range rpcsToTest {
+				errorPrefix := fmt.Sprintf("[request %d/%q: rpc %q/%d/%q]",
+					requestIdx, requestProto.GetName(), group.Name, rpcIdx, rpcName)
+
+				// Ensure that we issue only the RPCs the test suite is expecting.
+				method, ok := restRPCs[rpcName]
+				if !ok {
+					t.Errorf("%s could not find client library method for this RPC", errorPrefix)
+					continue
+				}
+
+				response, err := method(context.Background(), requestProto)
+				if err != nil {
+					t.Errorf("%s error: %s", errorPrefix, err)
+				}
+				// Check for expected response.
+				if diff := cmp.Diff(response.GetInfo(), requestProto.GetInfo(), cmp.Comparer(proto.Equal)); diff != "" {
+					t.Errorf("%s unexpected response: got=-, want=+:%s\n------------------------------\n",
+						errorPrefix, diff)
+				}
+			}
+		}
+	}
+}
+
+// getComplianceSuite returns the ComplianceSuite read and parsed from the appropriate location
+// (current directory if available, otherwise the module path).
+func getComplianceSuite() (*genprotopb.ComplianceSuite, error) {
+	filePath, err := getComplianceSuiteFile()
+	if err != nil {
+		return nil, err
+	}
+
+	complianceSuiteJSON, err := ioutil.ReadFile(filePath)
+	if err != nil {
+		return nil, fmt.Errorf("could not open suite file %q", filePath)
+	}
+
+	suite := &genprotopb.ComplianceSuite{}
+	if err := protojson.Unmarshal(complianceSuiteJSON, suite); err != nil {
+		return nil, fmt.Errorf("error unmarshalling from json %s:\n   file: %s\n   input was: %s", err, filePath, complianceSuiteJSON)
+	}
+	return suite, nil
+}
+
+// getComplianceSuiteFile returns the path to the compliance_suite.json file at the current
+// directory, if a file with that name is found there, or in the Showcase module's path
+// optherwise. If such a filename cannot be found in either place, this returns an error.
+func getComplianceSuiteFile() (string, error) {
+	const fileName = "compliance_suite.json"
+	pathInShowcase := path.Join("github.com", "googleapis", "gapic-showcase@v"+showcaseSemver, "server", "services", fileName)
+
+	// Return the file in the current directory, if it exists there.
+	currentDir, err := os.Getwd()
+	if err != nil {
+		return "", fmt.Errorf("could not determine current directory: %s", err)
+	}
+
+	filePath := path.Join(currentDir, fileName)
+	if _, err := os.Stat(filePath); err == nil {
+		return filePath, nil
+	}
+
+	// Return the file in the installed module.
+	goCmd := exec.Command("go", "env", "GOMODCACHE")
+	output, err := goCmd.Output()
+	if err != nil {
+		return "", fmt.Errorf("could not determine GOMODCACHE: %s", err)
+	}
+
+	filePath = path.Join(strings.TrimSpace(string(output)), pathInShowcase)
+	if _, err := os.Stat(filePath); err != nil {
+		return "", fmt.Errorf("could not determine location of %q; %s", fileName, err)
+	}
```

Comment:
More out of curiosity, this is to load the compliance suite file?

Reply:
I believe so.